pyprql/polars_sub_namespace/prql.py

```diff
@@ -0,0 +1,21 @@
+import polars as pl
+import prqlc
+
+
+@pl.api.register_dataframe_namespace("prql")
+@pl.api.register_lazyframe_namespace("prql")
+class PrqlSubNamespace:
+    def __init__(self, df: pl.DataFrame | pl.LazyFrame):
+        self._df = df
+
+    def query(
+        self, prql_query: str, *, table_name: str | None = None
+    ) -> pl.DataFrame | pl.LazyFrame:
```

Comment:
I think this works fine, but I couldn't figure out how to write the type hint here. Obviously the return value changes depending on the df type, but is there any way to indicate that?
(Perhaps I should inherit the class and define `PrqlDataFrameSubNs` and `PrqlLazyFrameSubNs`?)

Reply:
Yes if we do something like this:

```
T_Df = TypeVar(pl.DataFrame, pl.LazyFrame)

class PrqlSubNamespace[T_Df]:
    def query(...) -> T_Df
```

...that should work.

Here's an example of a similar impl:
https://github.com/pydata/xarray/blob/d64460795e406bc4a998e2ddae0054a1029d52a9/xarray/core/parallel.py#L229-L235
https://github.com/pydata/xarray/blob/9eb180b7b3df869c21518a17c6fa9eb456551d21/xarray/core/types.py#L154

Reply:
Thanks! I will try that.